### ISSUE TITLE
search failed with cluster

### ISSUE THREAD

**Issue by chenlizhao** [dengliming/redis-modules-java#43]
centos7.9, redis-6.2.5, I create a cluster with 6 nodes, and json set one document as:

RediSearchClient rediSearchClient = new RediSearchClient(config);
		RediSearch rediSearch = rediSearchClient.getRediSearch("testSearch");
		rediSearch.dropIndex();
		rediSearch.createIndex(new io.github.dengliming.redismodule.redisearch.index.schema.Schema()
			    .addField(new io.github.dengliming.redismodule.redisearch.index.schema.TextField("title"))
			    .addField(new io.github.dengliming.redismodule.redisearch.index.schema.TextField("content"))
			    .addField(new Field("age", FieldType.NUMERIC))
			    .addField(new Field("location", FieldType.GEO)));
		Map<String, Object> fields = new HashMap<>();
		fields.put("title", "Hi");
		fields.put("content", "OOOO");
		RedisJSONClient redisJSONClient = new RedisJSONClient(config);
		RedisJSON redisJSON = redisJSONClient.getRedisJSON();
		redisJSON.set("user1", SetArgs.Builder.create(".", GsonUtils.toJson(fields)));

and search without filter as:
SearchResult searchResult = rediSearch.search("*", new SearchOptions()
                .noStopwords()
                .language(RSLanguage.ENGLISH));
but the result is: 
{
total: 0,
documents: [ ]
}

what's wrong with my search?

**Comment by dengliming** [comment]
Currently. we don't support indexing JSON documents yet. see https://github.com/dengliming/redis-modules-java/issues/26

**Comment by dengliming** [comment]
@chenlizhao Can you try this? https://github.com/dengliming/redis-modules-java/blob/3017a4fe1cb2f8bca413f7ce4659094f35422720/redisearch/src/test/java/io/github/dengliming/redismodule/redisearch/RediSearchTest.java#L81-L97

**Comment by chenlizhao** [comment]
> @chenlizhao Can you try this?
> 
> https://github.com/dengliming/redis-modules-java/blob/3017a4fe1cb2f8bca413f7ce4659094f35422720/redisearch/src/test/java/io/github/dengliming/redismodule/redisearch/RediSearchTest.java#L81-L97

Yes, the search is ok for single node now, but for cluster, the search is still fail, for example, json of user john  is set on node1, but the schema is created on node2, when I search user john, the result is still empty.

**Comment by dengliming** [comment]
@chenlizhao Can you provider your example?

**Comment by chenlizhao** [comment]
@dengliming  here is my code:

Config config = new Config();		
			config.useClusterServers().addNodeAddress("redis://10.200.5.119:7001", "redis://10.200.5.119:7002", "redis://10.200.5.119:7003",
					"redis://10.200.5.119:7004", "redis://10.200.5.119:7005", "redis://10.200.5.119:7006");
			RediSearchClient rediSearchClient = new RediSearchClient(config);
			RediSearch rediSearch = rediSearchClient.getRediSearch("userIdx");
rediSearch.dropIndex();
			rediSearch.createIndex(new io.github.dengliming.redismodule.redisearch.index.schema.Schema()
				    .addField(new io.github.dengliming.redismodule.redisearch.index.schema.TextField("$.title").attribute("title"))
				    .addField(new io.github.dengliming.redismodule.redisearch.index.schema.TextField("$.content").attribute("content"))
				    .addField(new Field("$.age", FieldType.NUMERIC))
				    .addField(new Field("$.location", FieldType.GEO)),
				    new IndexOptions().definition(new io.github.dengliming.redismodule.redisearch.index.IndexDefinition(io.github.dengliming.redismodule.redisearch.index.IndexDefinition.DataType.JSON)));
			Map<String, Object> fields = new HashMap<>();
			fields.put("title", "Hi");
			fields.put("content", "OOOO");
			fields.put("age", 2);
			RedisJSONClient redisJSONClient = new RedisJSONClient(config);
			RedisJSON redisJSON = redisJSONClient.getRedisJSON();
			redisJSON.set("user1", SetArgs.Builder.create("$", GsonUtils.toJson(fields)));
			fields.put("title", "test");
			redisJSON.set("user2", SetArgs.Builder.create("$", GsonUtils.toJson(fields)));
io.github.dengliming.redismodule.redisearch.search.SearchResult searchResult = rediSearch.search("*", new SearchOptions()
	                .noStopwords()
	                .language(RSLanguage.ENGLISH));

it should return both user1 and user2, but user2 is not set on the same node of user1, so the result only returns use1

**Comment by dengliming** [comment]
Hey. how did you build/run the module?  
see https://github.com/RediSearch/RediSearch/issues/2557

**Comment by chenlizhao** [comment]
when I enabled COORD=oss, the search result  contains duplicated  keys, what should I do?

**Comment by dengliming** [comment]
@chenlizhao Did you flushdb and retry again? Then paste your results here.

**Comment by chenlizhao** [comment]
I clean all db and recreate cluster, but the resut is still not ok as:

{
total: 3,
documents: [
{
id: "user1",
score: 1,
fields: {
$: "{"title":"Hi","content":"OOOO","age":2}"
}
},
{
id: "user2",
score: 1,
fields: {
$: "{"title":"test","content":"OOOO","age":2}"
}
},
{
id: "user2",
score: 1,
fields: {
$: "{"title":"test","content":"OOOO","age":2}"
}
}
]
}

there are two user2 in result.

**Comment by dengliming** [comment]
@chenlizhao Hey, May I ask why you closed this issue? have you solved your problem?

**Comment by chenlizhao** [comment]
@dengliming yes, aftet I updated to latest version of redisearch, it was ok, thx for your help!
